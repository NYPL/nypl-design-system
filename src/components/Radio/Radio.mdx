import { ArgTypes, Canvas, Description, Meta } from "@storybook/blocks";

import * as RadioStories from "./Radio.stories";
import Link from "../Link/Link";

<Meta of={RadioStories} />

# Radio

| Component Version | DS Version |
| ----------------- | ---------- |
| Added             | `0.22.0`   |
| Latest            | `1.5.0`    |

## Table of Contents

- {<Link href="#overview" target="_self">Overview</Link>}
- {<Link href="#component-props" target="_self">Component Props</Link>}
- {<Link href="#accessibility" target="_self">Accessibility</Link>}
- {<Link href="#browser-states" target="_self">Browser States</Link>}
- {<Link href="#helper-and-error-text" target="_self">Helper and Error Text</Link>}
- {<Link href="#label-using-jsx-elements" target="_self">Label Using JSX Elements</Link>}

## Overview

<Description of={RadioStories} />

This component renders a radio button form element. Note that all these examples
show the `Radio` button in isolation. We recommend to always use the `Radio`
component inside the Design System `RadioGroup` component. The `RadioGroup`
component will handle all the states and data management.

## Component Props

<Canvas of={RadioStories.Controls} />

<ArgTypes of={RadioStories.Controls} />

## Accessibility

The `Radio` component renders an `input` element with the `type="radio"` attribute.
This HTML element is accessible by default.

When `showLabel` is set to false, the `labelText` value will be set to the
`<input>`'s `aria-label` attribute.

Resources:

- [W3C ARIA radiogroup and radio example](https://www.w3.org/TR/2016/WD-wai-aria-practices-1.1-20160317/examples/radio/radio.html)
- [W3C Radio Button Example](https://www.w3.org/TR/2017/WD-wai-aria-practices-1.1-20170628/examples/radio/radio-1/radio-1.html)
- [MDN input type="radio"](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio)
- [Deque University Radio and Radio Group](https://dequeuniversity.com/library/aria/radio-and-radio-group)
- [Chakra UI Radio](https://chakra-ui.com/docs/components/form/radio)

## Browser States

<Canvas of={RadioStories.BrowserStates} />

## Helper and Error Text

<Canvas of={RadioStories.HelperAndErrorText} />

## JSX Element Labels

This is useful when you want to add dynamic content to the label or add
a layout to the label. View the `RadioGroup` documentation for this
usage.

<Canvas of={RadioStories.LabelUsingJSXElements} />
