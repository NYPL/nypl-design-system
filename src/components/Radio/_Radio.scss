.radio {
  -webkit-appearance: none;
  border: 2px solid $black;
  border-radius: 50%;
  display: inline-block;
  height: 1.375rem;
  outline: none;
  position: relative;
  vertical-align: middle;
  width: 1.375rem;

  &:before {
    border-radius: 100%;
    content: "";
    display: block;
    height: 0.75rem;
    left: 50%;
    position: absolute;
    top: 50%;
    transform: translate(-50%, -50%);
    width: 0.75rem;
  }

  &:checked {
    border-color: $focus;

    &:before {
      background-color: $focus;
    }
  }

  + label {
    display: inline-block;
    font-size: var(--font-size-0);
    font-weight: 300;
    margin-bottom: 0px;
    margin-left: var(--space-xs);
    vertical-align: middle;

    + .helper-text {
      margin-bottom: 0;
      margin-left: calc(22px + var(--space-xs));
      margin-top: var(--space-xs);
    }
  }

  &--error {
    border-color: $error;

    + label {
      color: $error;

      + .helper-text {
        color: $error;
      }
    }

    &:checked {
      border-color: $error;

      &:before {
        background-color: $error;
      }
    }
  }

  &:disabled {
    border-color: $gray-medium;

    &:before {
      background-color: $gray-medium;
    }

    + label {
      color: $gray-dark;
      font-style: italic;

      + .helper-text {
        color: $gray-dark;
        font-style: italic;
      }
    }
  }
}
