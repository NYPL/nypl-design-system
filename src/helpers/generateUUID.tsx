import { v4 as uuidv4 } from "uuid";

export default function generateUUID() {
  return uuidv4();
}
