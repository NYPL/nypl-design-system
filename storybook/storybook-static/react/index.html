<!doctype html><html lang="en"><head><meta charset="utf-8"><title>Storybook</title><meta name="viewport" content="width=device-width,initial-scale=1"><style>html, body {
    overflow: hidden;
    height: 100%;
    width: 100%;
    margin: 0;
    padding: 0;
  }</style><script>/* globals window */
  /* eslint-disable no-underscore-dangle */
  try {
    if (window.top !== window) {
      window.__REACT_DEVTOOLS_GLOBAL_HOOK__ = window.top.__REACT_DEVTOOLS_GLOBAL_HOOK__;
    }
  } catch (e) {
    // eslint-disable-next-line no-console
    console.warn('unable to connect to top frame for connecting dev tools');
  }</script><style>#root[hidden],
      #docs-root[hidden] {
        display: none !important;
      }</style></head><body><div id="root"></div><div id="docs-root"></div><script>window['DOCS_MODE'] = false;</script><script src="runtime~main.00bd9d033d7a0b972fc9.bundle.js"></script><script src="vendors~main.d2b6aafef1a259d165c0.bundle.js"></script><script src="main.3bf71672a6dd35499004.bundle.js"></script></body></html>