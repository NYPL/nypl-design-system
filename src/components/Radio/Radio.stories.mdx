import {
  Meta,
  Story,
  Canvas,
  ArgsTable,
  Preview,
  Description,
} from "@storybook/addon-docs/blocks";
import { withDesign } from "storybook-addon-designs";
import { getCategory } from "../../utils/componentCategories";

import Radio from "./Radio";
import { onChangeDefault } from "./Radio";

<Meta
  title={getCategory("Radio")}
  component={Radio}
  decorators={[withDesign]}
  parameters={{
    design: {
      type: "figma",
      url:
        "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Master?node-id=11895%3A742",
    },
  }}
  argTypes={{
    onChange: { table: { disable: true } },
    checked: { table: { disable: true } },
  }}
/>

# Radio

<Description of={Radio} />

This component renders a radio button form element. See below for configuration information. Note that the `id` property, while optional, will
cause an accessibility violation if none is present.

<Preview withToolbar>
  <Story
    name="Radio"
    args={{
      labelText: "Test Label",
      showLabel: true,
      id: "test_id",
      name: "test_name",
    }}
  >
    {args => <Radio {...args} />}
  </Story>
</Preview>

<ArgsTable story="Radio" />

## Checked

<Canvas>
  <Radio
    checked
    labelText="I am checked"
    name="testChecked"
    onChange={onChangeDefault}
    showLabel={true}
  ></Radio>
</Canvas>

## Focused

<Canvas>
  <Radio
    name="testFocus"
    labelText="Click or tab to the Radio to see its focus state"
    showLabel={true}
  ></Radio>
</Canvas>

## Errored

<Canvas>
  <Radio
    name="testErrored"
    errored
    showLabel={true}
    labelText="I am in an error state"
  ></Radio>
</Canvas>

## Disabled

<Canvas>
  <Radio
    name="testDisabled"
    disabled
    showLabel={true}
    labelText="I am disabled"
  ></Radio>
</Canvas>

## Helper Text

<Canvas>
  <Radio
    name="testHelperText"
    showLabel={true}
    labelText="I have helper text"
    helperText="I am the helper text for this Radio"
  ></Radio>
</Canvas>

## Error Text

<Canvas>
  <Radio
    name="testErrorText"
    showLabel={true}
    errored
    labelText="I have error text"
    errorText="I am the error text for this Radio"
  ></Radio>
</Canvas>
