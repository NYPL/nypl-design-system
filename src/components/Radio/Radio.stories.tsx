import { Box, HStack, VStack } from "@chakra-ui/react";
import type { Meta, StoryObj } from "@storybook/react";
import { withDesign } from "storybook-addon-designs";

import Heading from "../Heading/Heading";
import Radio from "./Radio";

const meta: Meta<typeof Radio> = {
  title: "Components/Form Elements/Radio",
  component: Radio,
  decorators: [withDesign],
  argTypes: {
    className: { control: false },
    id: { control: false },
    isChecked: { control: false },
    isDisabled: {
      table: { defaultValue: { summary: false } },
    },
    isInvalid: {
      table: { defaultValue: { summary: false } },
    },
    isRequired: {
      table: { defaultValue: { summary: false } },
    },
    key: { table: { disable: true } },
    onChange: { control: false },
    ref: { table: { disable: true } },
    showHelperInvalidText: {
      table: { defaultValue: { summary: true } },
    },
    showLabel: {
      table: { defaultValue: { summary: true } },
    },
  },
};

export default meta;
type Story = StoryObj<typeof Radio>;

/**
 * Main Story for the Radio component. This must contains the `args`
 * and `parameters` properties in this object.
 */
export const Controls: Story = {
  args: {
    className: undefined,
    helperText: "This is the helper text!",
    id: "radio-id",
    invalidText: "This is the error text :(",
    isChecked: undefined,
    isDisabled: false,
    isInvalid: false,
    isRequired: false,
    labelText: "Test Label",
    name: "test_name",
    onChange: undefined,
    showHelperInvalidText: true,
    showLabel: true,
    value: "1",
  },
  render: (args) => <Radio {...args} />,
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Master?node-id=11895%3A742",
    },
    jest: ["Radio.test.tsx"],
  },
};

// The following are additional Radio example Stories.
export const BrowserStates: Story = {
  render: () => (
    <VStack align="left" spacing="l">
      <Box>
        <Heading level="three">Checked</Heading>
        <Radio labelText="I am checked" id="checked" isChecked value="1" />
      </Box>
      <Box>
        <Heading level="three">Focused</Heading>
        <Radio
          id="focused"
          labelText="Click or tab to the Radio to see its focus state"
        />
      </Box>
      <Box>
        <Heading level="three">Errored</Heading>
        <HStack>
          <Radio id="invalid" isInvalid labelText="I am in an error state" />
          <Radio
            id="invalid-checked"
            isInvalid
            isChecked
            labelText="I am checked in an error state"
          />
        </HStack>
      </Box>
      <Box>
        <Heading level="three">Disabled</Heading>
        <HStack>
          <Radio id="disabled" isDisabled labelText="I am disabled" />
          <Radio
            id="disabled-checked"
            isDisabled
            isChecked
            labelText="I am checked and disabled"
          />
        </HStack>
      </Box>
    </VStack>
  ),
};

export const HelperAndErrorText: Story = {
  render: () => (
    <VStack align="left" spacing="l">
      <Box>
        <Heading level="three">Helper Text</Heading>
        <Radio
          helperText="I am the helper text for this Radio"
          id="help-text"
          labelText="I have helper text"
          name="testHelperText"
        />
      </Box>
      <Box>
        <Heading level="three">Error Text</Heading>
        <Radio
          id="invalid-text"
          invalidText="I am the error text for this Radio"
          isInvalid
          labelText="I have error text"
          name="testErrorText"
        />
      </Box>
    </VStack>
  ),
};

export const LabelUsingJSXElements: Story = {
  render: () => (
    <Radio
      id="jsx-label"
      labelText={<span>Arts</span>}
      name="jsxElementLabel"
    />
  ),
};
