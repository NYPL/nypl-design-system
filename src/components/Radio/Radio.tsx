import * as React from "react";
import bem from "../../utils/bem";
import HelperErrorText from "../HelperErrorText/HelperErrorText";
import Label from "../Label/Label";
import generateUUID from "../../helpers/generateUUID";

export interface RadioProps {
  /** The radio button's label.  This will serve as the text content for a `<label>` element if `showlabel` is true, or an "aria-label" if `showLabel` is false. */
  labelText: string;
  /** Additional attributes to pass to the `<input>` tag */
  attributes?: { [key: string]: any };
  /** When using the Radio as a "controlled" form element, you can specify the Radio's checked state using this prop. Learn more about controlled and uncontrolled form fields: https://goshakkk.name/controlled-vs-uncontrolled-inputs-react/ */
  checked?: boolean;
  /** className you can add in addition to 'input' */
  className?: string;
  /** Adds the 'disabled' prop to the input when true */
  disabled?: boolean;
  /** Helper for modifiers array; adds 'errored' styling */
  errored?: boolean;
  /** Optional string to populate the HelperErrorText for error state */
  errorText?: string;
  /** Optional string to populate the HelperErrorText for standard state */
  helperText?: string;
  /** ID that other components can cross reference for accessibility purposes */
  id?: string;
  /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
  modifiers?: string[];
  /** The name prop indicates into which group of radios this radio belongs.  If none is specified, 'default' will be used */
  name?: string;
  /** The action to perform on the `<input>`'s onChange function  */
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
  /** Offers the ability to show the radio's label onscreen or hide it. Refer to the `labelText` property for more information. */
  showLabel: boolean;
  /** Populates the value of the input */
  value?: string;
}

export const onChangeDefault = () => {
  return;
};

const Radio = React.forwardRef<HTMLInputElement, RadioProps>((props, ref?) => {
  const {
    checked,
    className,
    disabled,
    errored,
    errorText,
    helperText,
    id,
    labelText,
    name,
    showLabel,
    value,
  } = props;

  const attributes = props.attributes || {};
  const modifiers = props.modifiers ? props.modifiers : [];
  const radioID = id || generateUUID();
  const onChange = props.onChange || onChangeDefault;

  const footnote = errored ? errorText : helperText;

  if (!showLabel) {
    attributes["aria-label"] =
      labelText && footnote ? `${labelText} - ${footnote}` : labelText;
  } else {
    if (footnote) attributes["aria-describedby"] = `${radioID}-helperText`;
  }

  if (errored) {
    modifiers.push("error");
  }

  return (
    <>
      <input
        {...(checked !== undefined
          ? {
              checked: checked,
              onChange: onChange,
            }
          : {
              defaultChecked: false,
            })}
        className={bem("radio", modifiers, "", [className])}
        disabled={disabled}
        id={radioID}
        name={name || "default"}
        ref={ref}
        type="radio"
        value={value}
        onChange={onChange}
        {...attributes}
      />
      {labelText && showLabel && <Label htmlFor={radioID}>{labelText}</Label>}
      {footnote && showLabel && (
        <HelperErrorText isError={errored} id={id + `-helperText`}>
          {footnote}
        </HelperErrorText>
      )}
    </>
  );
});

Radio.displayName = "Radio";

export default Radio;
