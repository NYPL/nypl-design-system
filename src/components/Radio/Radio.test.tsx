import { expect } from "chai";
import * as Enzyme from "enzyme";
import * as React from "react";
import { stub, spy } from "sinon";
import generateUUID from "../../helpers/generateUUID";

import Radio from "./Radio";

describe("Radio Button", () => {
  let container;
  let changeHandler;
  let clickHander;
  let generateUUIDSpy;

  before(() => {
    clickHander = stub();
    changeHandler = stub();
    generateUUIDSpy = spy(generateUUID);
    container = Enzyme.mount(
      <Radio
        id="inputID"
        attributes={{ onClick: clickHander }}
        onChange={changeHandler}
        labelText="Test Label"
        showLabel={false}
      ></Radio>
    );
  });

  it("Renders with appropriate 'aria-label' attribute and value when 'showLabel' prop is set to false", () => {
    const labelText = container.prop("labelText");
    expect(container.find("input").prop("aria-label")).to.equal(labelText);
  });

  it("Renders without crashing", () => {
    expect(container.find("input").exists()).to.equal(true);
  });

  it("The radio element is an input with type='radio'", () => {
    expect(container.find("input").prop("type")).to.equal("radio");
  });

  it("The radio element's ID is set properly using the value passed to it.", () => {
    expect(container.find("input").prop("id")).to.equal("inputID");
  });

  it("Allows user to pass in additional attributes", () => {
    container.simulate("click");
    expect(clickHander.callCount).to.equal(1);
  });

  it("Changing the value calls the onChange handler", () => {
    container.find("input").simulate("change", { target: { value: "Hello" } });
    expect(changeHandler.callCount).to.equal(1);
  });

  it("Renders with appropriate 'aria-label' attribute and value when 'showLabel' prop is set to false and 'helperText' has been passed", () => {
    container = Enzyme.mount(
      <Radio
        //add other props here
        id="test_id"
        labelText="Hello"
        showLabel={false}
        helperText="The helper text."
      ></Radio>
    );
    const labelText = container.prop("labelText");
    const helperText = container.prop("helperText");
    expect(container.find("input").prop("aria-label")).to.equal(
      `${labelText} - ${helperText}`
    );
  });

  it("Renders with label", () => {
    container = Enzyme.mount(
      <Radio
        //add other props here
        id="test_id"
        labelText="Hello"
        showLabel={true}
      ></Radio>
    );
    expect(container.find("label").exists()).to.equal(true);
    expect(container.find("input").props()).not.to.have.property("aria-label");
    const radioId = container.prop("id");
    expect(container.find("label").prop("htmlFor")).to.equal(radioId);
  });

  it("Renders with visible helper text", () => {
    container = Enzyme.mount(
      <Radio
        //add other props here
        id="test_id"
        labelText="Hello"
        showLabel={true}
        helperText="The helper text."
        errorText="The error text."
        errored={false}
      ></Radio>
    );
    expect(container.find(".helper-text").text()).to.equal("The helper text.");
  });

  it("Renders with visible error text", () => {
    container = Enzyme.mount(
      <Radio
        //add other props here
        id="test_id"
        labelText="Hello"
        showLabel={true}
        helperText="The helper text."
        errorText="The error text."
        errored={true}
      ></Radio>
    );
    expect(container.find(".helper-text").text()).to.equal("The error text.");
  });

  it("Calls a UUID generation method if no ID is passed as a prop", () => {
    container = Enzyme.mount(
      <Radio labelText="Hello" showLabel={true}></Radio>
    );
    expect(container.find("input").props()).to.have.property("id");

    expect(generateUUIDSpy.called);
  });

  it("The 'checked' attribute can set properly", () => {
    const onChange = stub();
    const container = Enzyme.mount(
      <Radio
        id="inputID-attributes"
        labelText="Hello"
        showLabel={false}
        attributes={{
          checked: true,
          "aria-checked": true,
          onChange,
        }}
      ></Radio>
    );

    const input = container.find("input");

    expect(input.prop("checked")).to.equal(true);
    expect(input.prop("aria-checked")).to.equal(true);
  });

  it("Passes the ref to the input element", () => {
    const ref = React.createRef<HTMLInputElement>();
    const container = Enzyme.mount(
      <Radio
        id="inputID-attributes"
        ref={ref}
        labelText="Hello"
        showLabel={false}
      ></Radio>
    );
    expect(container.find("input").instance()).to.equal(ref.current);
  });

  it("Renders HTML attributes passed through the `attributes` prop", () => {
    const onChangeSpy = stub();
    const onBlurSpy = stub();
    container = Enzyme.mount(
      <Radio
        id="inputID-attributes"
        attributes={{
          onChange: onChangeSpy,
          onBlur: onBlurSpy,
          tabIndex: 0,
        }}
        labelText="Hello"
        showLabel={true}
      ></Radio>
    );
    expect(container.find("input").prop("tabIndex")).to.equal(0);
    expect(onChangeSpy.callCount).to.equal(0);
    container.find("input").simulate("change");
    expect(onChangeSpy.callCount).to.equal(1);
    expect(onBlurSpy.callCount).to.equal(0);
    container.find("input").simulate("blur");
    expect(onBlurSpy.callCount).to.equal(1);
  });
});
