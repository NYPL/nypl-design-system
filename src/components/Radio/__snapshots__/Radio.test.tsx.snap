// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Button renders the UI snapshot correctly 1`] = `
<label
  className="chakra-radio css-13p0l12"
>
  <input
    checked={false}
    className="chakra-radio__input"
    disabled={false}
    id="inputID"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    required={false}
    style={
      Object {
        "border": "0px",
        "clip": "rect(0px, 0px, 0px, 0px)",
        "height": "1px",
        "margin": "-1px",
        "overflow": "hidden",
        "padding": "0px",
        "position": "absolute",
        "whiteSpace": "nowrap",
        "width": "1px",
      }
    }
    type="radio"
  />
  <span
    aria-hidden={true}
    className="css-ssalds"
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
  />
  <span
    className="chakra-radio__label css-1y8kf23"
    onMouseDown={[Function]}
    onTouchStart={[Function]}
  >
    Test Label
  </span>
</label>
`;

exports[`Radio Button renders the UI snapshot correctly 2`] = `
<label
  className="chakra-radio css-13p0l12"
  data-checked=""
>
  <input
    checked={true}
    className="chakra-radio__input"
    disabled={false}
    id="radio-checked"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    required={false}
    style={
      Object {
        "border": "0px",
        "clip": "rect(0px, 0px, 0px, 0px)",
        "height": "1px",
        "margin": "-1px",
        "overflow": "hidden",
        "padding": "0px",
        "position": "absolute",
        "whiteSpace": "nowrap",
        "width": "1px",
      }
    }
    type="radio"
  />
  <span
    aria-hidden={true}
    className="css-ssalds"
    data-checked=""
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
  />
  <span
    className="chakra-radio__label css-1y8kf23"
    data-checked=""
    onMouseDown={[Function]}
    onTouchStart={[Function]}
  >
    Test Label
  </span>
</label>
`;

exports[`Radio Button renders the UI snapshot correctly 3`] = `
<label
  className="chakra-radio css-13p0l12"
>
  <input
    aria-required={true}
    checked={false}
    className="chakra-radio__input"
    disabled={false}
    id="radio-required"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    required={true}
    style={
      Object {
        "border": "0px",
        "clip": "rect(0px, 0px, 0px, 0px)",
        "height": "1px",
        "margin": "-1px",
        "overflow": "hidden",
        "padding": "0px",
        "position": "absolute",
        "whiteSpace": "nowrap",
        "width": "1px",
      }
    }
    type="radio"
  />
  <span
    aria-hidden={true}
    className="css-ssalds"
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
  />
  <span
    className="chakra-radio__label css-1y8kf23"
    onMouseDown={[Function]}
    onTouchStart={[Function]}
  >
    Test Label
  </span>
</label>
`;

exports[`Radio Button renders the UI snapshot correctly 4`] = `
<label
  className="chakra-radio css-13p0l12"
  data-invalid=""
>
  <input
    aria-invalid={true}
    checked={false}
    className="chakra-radio__input"
    disabled={false}
    id="radio-invalid"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    required={false}
    style={
      Object {
        "border": "0px",
        "clip": "rect(0px, 0px, 0px, 0px)",
        "height": "1px",
        "margin": "-1px",
        "overflow": "hidden",
        "padding": "0px",
        "position": "absolute",
        "whiteSpace": "nowrap",
        "width": "1px",
      }
    }
    type="radio"
  />
  <span
    aria-hidden={true}
    className="css-ssalds"
    data-invalid=""
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
  />
  <span
    className="chakra-radio__label css-1y8kf23"
    data-invalid=""
    onMouseDown={[Function]}
    onTouchStart={[Function]}
  >
    Test Label
  </span>
</label>
`;

exports[`Radio Button renders the UI snapshot correctly 5`] = `
<label
  className="chakra-radio css-13p0l12"
  data-disabled=""
>
  <input
    aria-disabled={true}
    checked={false}
    className="chakra-radio__input"
    disabled={true}
    id="radio-disabled"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    required={false}
    style={
      Object {
        "border": "0px",
        "clip": "rect(0px, 0px, 0px, 0px)",
        "height": "1px",
        "margin": "-1px",
        "overflow": "hidden",
        "padding": "0px",
        "position": "absolute",
        "whiteSpace": "nowrap",
        "width": "1px",
      }
    }
    type="radio"
  />
  <span
    aria-hidden={true}
    className="css-ssalds"
    data-disabled=""
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
  />
  <span
    className="chakra-radio__label css-1y8kf23"
    data-disabled=""
    onMouseDown={[Function]}
    onTouchStart={[Function]}
  >
    Test Label
  </span>
</label>
`;

exports[`Radio Button renders the UI snapshot correctly 6`] = `
<label
  className="chakra-radio css-13p0l12"
>
  <input
    checked={false}
    className="chakra-radio__input"
    disabled={false}
    id="jsxLabel"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    required={false}
    style={
      Object {
        "border": "0px",
        "clip": "rect(0px, 0px, 0px, 0px)",
        "height": "1px",
        "margin": "-1px",
        "overflow": "hidden",
        "padding": "0px",
        "position": "absolute",
        "whiteSpace": "nowrap",
        "width": "1px",
      }
    }
    type="radio"
    value="arts"
  />
  <span
    aria-hidden={true}
    className="css-ssalds"
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
  />
  <span
    className="chakra-radio__label css-1y8kf23"
    onMouseDown={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="css-k008qs"
    >
      <span>
        Arts
      </span>
      <div
        className="css-17xejub"
      />
      <span>
        4
      </span>
    </div>
  </span>
</label>
`;
